lib/_http_outgoing.js

```diff
@@ -234,12 +237,18 @@ OutgoingMessage.prototype._send = function _send(data, encoding, callback) {
         (encoding === 'utf8' || encoding === 'latin1' || !encoding)) {
       data = this._header + data;
     } else {
-      this.output.unshift(this._header);
-      this.outputEncodings.unshift('latin1');
-      this.outputCallbacks.unshift(null);
-      this.outputSize += this._header.length;
-      if (typeof this._onPendingData === 'function')
-        this._onPendingData(this._header.length);
+      var header = this._header;
+      if (this.output.length === 0) {
+        this.output = [header];
+        this.outputEncodings = ['latin1'];
```

Comment:
General note on how we handle the response body. Browsers treat `'charset=latin1'` as `windows-1252`, which is problematic. Example:
```js
const headers =
  'HTTP/1.1 200 OK\r\n' +
  'Content-Type: text/html; charset=latin1\r\n' +
  'Content-Length: 2\r\n\r\n' +
  // Technically 0x83 is a control code, but browsers don't see it that way.
  '\u0083\n';
const data = Buffer.from(headers, 'latin1');

require('net').createServer(c => c.end(data)).listen(8778);
```
Hit that with the browser and you'll see `ƒ`, not a control character symbol.